File: TDAT3020 Sikkerhet i programvare og nettverk/ovinger/oving04 - low level code/oppg2/main.cpp
#include <iostream>
#include <string>
#include <regex>

using namespace std;

string replace(string input)
{
    string newString;
    newString = regex_replace(input, regex("&"), "&amp;");
    newString = regex_replace(newString, regex("<"), "&lt;");
    newString = regex_replace(newString, regex(">"), "&gt;");
    return newString;
}

int main()
{
    string input;
    cout << "Enter text: "; 
    cin >> input;

    string replacedText = replace(input);

    cout << "output: " << replacedText << endl;
}

File: TDAT3020 Sikkerhet i programvare og nettverk/ovinger/oving04 - low level code/oppg2/main.rs
fn replace(mut input: String) -> String
{
    let mut newString=String::new();

    newString = input.replace("&", "&amp;");
    newString = newString.replace("<", "&lt;");
    newString = newString.replace(">", "&gt;");

    newString
}

fn main()
{
    use std::io::{stdin,stdout,Write};
    let mut s=String::new();

    print!("Enter text: ");
    let _=stdout().flush();
    stdin().read_line(&mut s).expect("Did not enter a correct string");
    
    if let Some('\n')=s.chars().next_back() {
        s.pop();
    }
    if let Some('\r')=s.chars().next_back() {
        s.pop();
    }

    let mut output = replace(s);

    println!("output: {:?}",output);
}